### PR TITLE
[Kernel][Standalone] Consolidate testing infrastructure that uses the GoldenTables project and other related clean-up

### PR DESCRIPTION
**Pull request by allisonport-db**
## Description

- Updates the `goldenTables` project to include utils that exposes the resource file path. Updates other projects to depend on it and use that for getting the golden table file paths.
  - Moves the resources to `src/main/resources` so that other projects can depend on the `goldenTables` compile conf and skip all the test dependencies
- Updates `goldenTables` to use the latest `delta-spark` version (just fixes a couple compile errors)
- Update the kernelDefault `DeletionVectorSuite`
   - Update `"end-to-end usage: reading partitioned dv table with checkpoint"` to use a golden table since that's how the resource table was originally generated. Adds this table to GoldenTables and removes the no longer used test resource table.
   - Update the test resource `"dv-with-columnmapping"` to actually have DV enabled (previously it mistakenly wasn't).

## How was this patch tested?

Testing change.

## Does this PR introduce _any_ user-facing changes?

No.

### CODE REVIEW

**Review by vkorukanti — APPROVED** [review]
LGTM. Few comments.

1) Delete the `.crc` files from DV tables?
2) Can you add the code for generating the DV test tables that were moved to golden tables module to https://github.com/delta-io/delta/blob/master/connectors/golden-tables/src/test/scala/io/delta/golden/GoldenTables.scala?
3) [Minor] Should we move the golden tables modules out of connectors? If that is preferred, may be do it in a separate PR?

**Comment by allisonport-db** [comment]
@vkorukanti 

(1) Removed the .crc files from the "dv-with-columnmapping" tables, but not from the golden table generated tables since those would just be regenerated

(2) I did that here https://github.com/delta-io/delta/pull/1887/files#diff-7f4dd956fba89d1268005d7dad5de577e341e43b14875e400ddc21723ae6d9c3R836. The table added to the golden table module was generated using that code (not copied over from the kernelDefault resources).

(3) I agree about a separate PR if we do it. I think it doesn't really matter and I thought we might consider moving all the modules in `connectors/` out of the inner nesting eventually anyways?

File: build.sbt [hunk 1]
```diff
@@ -770,14 +773,16 @@ lazy val compatibility = (project in file("connectors/oss-compatibility-tests"))
   )
  */
 
-lazy val goldenTables = (project in file("connectors/golden-tables")) settings (
+lazy val goldenTables = (project in file("connectors/golden-tables"))
+  // TODO: do we want to depend on latest released version or just directly on proj?
```
Comment (allisonport-db) [inline build.sbt:777]: @vkorukanti Do you have an opinion here?

**Review by vkorukanti — APPROVED** [review]
lgtm

**Pull request closed without merging**